### PR TITLE
[java] Fix #10080 Handle null parameters in TracedCommandExecutor

### LINKED ISSUE #10080: [🐛 Bug]: TracedCommandExecutor throws NPE for driver.switchTo().defaultContent() method for RemoteWebDriver
### What happened?

TracedCommandExecutor throws NPE for `driver.switchTo().defaultContent()` method for RemoteWebDriver
```
Caused by: java.lang.NullPointerException
	at org.openqa.selenium.remote.TracedCommandExecutor.execute(TracedCommandExecutor.java:47)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)
```
It fails [here](https://github.com/SeleniumHQ/selenium/blob/24a9c50fedecfea1216480c89e016552b539f271/java/src/org/openqa/selenium/remote/TracedCommandExecutor.java#L47) because 
`driver.switchTo().defaultContent() ` calls `DriverCommand.SWITCH_TO_FRAME(null)` and command parameter `null`. It causes  NPE on `parameter.getValue().toString()`

image

### How can we reproduce the issue?

```shell
@Test
    public void foo() throws MalformedURLException {
        // You have to setup some Selenium grid to use RemoteWebDriver
        ChromeOptions chromeOptions = new ChromeOptions();
        RemoteWebDriver driver = null;
        try {
            driver = new RemoteWebDriver(new URL("http://localhost:4444/wd/hub"), chromeOptions);
            driver.get("https://www.w3schools.com/html/html_iframe.asp");
            driver.switchTo().frame(0);
            driver.switchTo().defaultContent();
        } catch (Throwable e) {
            throw e;
        } finally {
            if (driver != null) {
                driver.quit();
            }
        }
    }
```

### Relevant log output

```shell
System info: host: 'bo-MS-7B86', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '5.11.0-40-generic', java.version: '1.8.0_302'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
Command: [6adcb1735cb29eb0dccefcc24f1dd320, switchToFrame {id=null}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 96.0.4664.45, chrome: ..., javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, proxy: Proxy(), setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: 6adcb1735cb29eb0dccefcc24f1dd320
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:583)
	at org.openqa.selenium.remote.RemoteWebDriver$RemoteTargetLocator.defaultContent(RemoteWebDriver.java:1042)
	at com.example.demo_selenied.MainPageTest.iframeTest2(MainPageTest.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    ...
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at org.openqa.selenium.remote.TracedCommandExecutor.execute(TracedCommandExecutor.java:47)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:559)
	...
```

### Operating System

Any

### Selenium version

4.1.0

### What are the browser(s) and version(s) where you see this issue?

Any remote

### What are the browser driver(s) and version(s) where you see this issue?

Any

### Are you using Selenium Grid?

Selenoid

### PR DESCRIPTION
**Pull request by BorisOsipov**
### Description
Bug fix for #10080
Converts null param value to "null" string.

Closes #10080
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by BorisOsipov** [comment]
Test failures seems unrelated to this changes

File: java/src/org/openqa/selenium/remote/TracedCommandExecutor.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public Response execute(Command command) throws IOException {
       Map<String, ?> parameters = command.getParameters();
       if (parameters != null && parameters.size() > 0) {
         for (Map.Entry<String, ?> parameter : parameters.entrySet()) {
-          commandSpan.setAttribute("parameter." + parameter.getKey(), parameter.getValue().toString());
+          commandSpan.setAttribute("parameter." + parameter.getKey(), String.valueOf(parameter.getValue()));
```
Comment (dacloutier-logmein) [inline java/src/org/openqa/selenium/remote/TracedCommandExecutor.java]: if `parameter.getValue()` is null, that will still throw a Null Pointer, unless mistaken.

if you're looking at ending up with `"null"` you'd be better of with :

`(String) parameter.getValue()`

File: java/test/org/openqa/selenium/remote/TracedCommandExecutorTest.java [hunk 2]
```diff
@@ -76,6 +76,22 @@ public void canCreateSpanWithAllAttributes() throws IOException {
...
+    SessionId sessionId = new SessionId(UUID.randomUUID());
+    Map<String, Object> parameters = new HashMap<>();
+    parameters.put("param1", null);
+    Command command = new Command(sessionId, "findElement", parameters);
+
+    tracedCommandExecutor.execute(command);
+
+    verify(span, times(1)).setAttribute("sessionId", sessionId.toString());
+    verify(span, times(1)).setAttribute("command", "findElement");
+    verify(span, times(1)).setAttribute("parameter.param1", String.valueOf((Object) null));
```
Comment (dacloutier-logmein) [inline java/test/org/openqa/selenium/remote/TracedCommandExecutorTest.java]: Subtlety between the test and the code above is the cast to `Object` that is done.

forgive me for using https://www.tutorialspoint.com/compile_java_online.php
but is was the quickest way for me to _test_ this as I am not on my laptop.

```
public class HelloWorld{
     public static void main(String []args){
        System.out.println("Hello World");
        System.out.println(String.valueOf((Object) null));
        System.out.println((String) null);
        System.out.println(String.valueOf((null)));
     }
}
```

yields:
```
Hello World
null
null
Exception in thread "main" java.lang.NullPointerException
	at java.lang.String.<init>(String.java:166)
	at java.lang.String.valueOf(String.java:3008)
	at HelloWorld.main(HelloWorld.java:7)
```

Comment (BorisOsipov) [reply java/test/org/openqa/selenium/remote/TracedCommandExecutorTest.java]: Thx. Removed useless cast and code.

File: java/src/org/openqa/selenium/remote/TracedCommandExecutor.java (hunk 1 above)
Comment (BorisOsipov) [reply java/src/org/openqa/selenium/remote/TracedCommandExecutor.java]: >if parameter.getValue() is null, that will still throw a Null Pointer, unless mistaken.

but test was green :)

ok let's make it clear with `Objects.toString(parameter.getValue(), "null")`

File: java/src/org/openqa/selenium/remote/TracedCommandExecutor.java [hunk 3]
```diff
@@ -44,7 +45,7 @@ public Response execute(Command command) throws IOException {
       Map<String, ?> parameters = command.getParameters();
       if (parameters != null && parameters.size() > 0) {
         for (Map.Entry<String, ?> parameter : parameters.entrySet()) {
-          commandSpan.setAttribute("parameter." + parameter.getKey(), parameter.getValue().toString());
+          commandSpan.setAttribute("parameter." + parameter.getKey(), Objects.toString(parameter.getValue(), "null"));
```
Comment (dacloutier-logmein) [inline java/src/org/openqa/selenium/remote/TracedCommandExecutor.java:48]: 👍

**Review by dacloutier-logmein — APPROVED** [review]

**Review by diemol — APPROVED** [review]
Thank you, @BorisOsipov! Congrats on your first contribution! 🎉 

Thank you for your help reviewing @dacloutier-logmein!

**Comment by pkoenig10** [comment]
When can we expect a new release with this fix?

**Pull request merged**